timelock-server/src/main/java/com/palantir/atlasdb/timelock/lock/BlockingTimeLimitedLockService.java

```diff
@@ -0,0 +1,223 @@
+                ImmutableLockRequestSpecification.of("lockWithFullLockResponse", client.getClientId(), request));
+    }
+
+    @Override
+    public boolean unlockSimple(SimpleHeldLocksToken token) {
+        return delegate.unlockSimple(token);
+    }
+
+    @Override
+    public boolean unlockAndFreeze(HeldLocksToken token) {
+        return delegate.unlockAndFreeze(token);
+    }
+
+    @Override
+    public Set<HeldLocksToken> getTokens(LockClient client) {
+        return delegate.getTokens(client);
+    }
+
+    @Override
+    public Set<HeldLocksToken> refreshTokens(Iterable<HeldLocksToken> tokens) {
+        return delegate.refreshTokens(tokens);
+    }
+
+    @Nullable
+    @Override
+    public HeldLocksGrant refreshGrant(HeldLocksGrant grant) {
+        return delegate.refreshGrant(grant);
+    }
+
+    @Nullable
+    @Override
+    public HeldLocksGrant refreshGrant(BigInteger grantId) {
+        return delegate.refreshGrant(grantId);
+    }
+
+    @Override
+    public HeldLocksGrant convertToGrant(HeldLocksToken token) {
+        return delegate.convertToGrant(token);
+    }
+
+    @Override
+    public HeldLocksToken useGrant(LockClient client, HeldLocksGrant grant) {
+        return delegate.useGrant(client, grant);
+    }
+
+    @Override
+    public HeldLocksToken useGrant(LockClient client, BigInteger grantId) {
+        return delegate.useGrant(client, grantId);
+    }
+
+    @Override
+    public LockServerOptions getLockServerOptions() {
+        return delegate.getLockServerOptions();
+    }
+
+    @Override
+    public long currentTimeMillis() {
+        return delegate.currentTimeMillis();
+    }
+
+    @Override
+    public void logCurrentState() {
+        delegate.logCurrentState();
+    }
+
+    private <T> T callWithTimeLimit(Callable<T> callable, LockRequestSpecification specification)
+            throws InterruptedException {
+        try {
+            return timeLimiter.callWithTimeout(callable, blockingTimeLimitMillis, TimeUnit.MILLISECONDS, true);
+        } catch (InterruptedException e) {
+            // In this case, the thread was interrupted for some other reason, perhaps because we lost leadership.
+            log.warn("Lock service was interrupted when servicing {} for client \"{}\"; request was {}",
+                    specification.method(),
+                    specification.client(),
+                    specification.lockRequest(),
+                    e);
+            throw e;
+        } catch (UncheckedTimeoutException e) {
+            // This is the legitimate timeout case we're trying to catch.
+            String errorMessage = String.format(
+                    "Lock service timed out after %s milliseconds when servicing %s for client \"%s\"; request was %s",
+                    blockingTimeLimitMillis,
+                    specification.method(),
+                    specification.client(),
+                    specification.lockRequest());
+            log.warn(errorMessage, e);
```

Comment:
It doesn't indicate a problem, we expect them to time out occasionally.  Seeing a small number of these in the logs isn't an issue but maybe something we'd investigate to find ways to optimise products.  This points to INFO.

If this is happening a huge number of times then it is a problem, but I expect this becomes clear from the volume.

If we downgrade to info you should also change the variable to message rather than errorMessage, otherwise you're just moving the alarm to when we read the code!